is3106backend/users/views.py

```diff
@@ -123,13 +114,17 @@ def protected_user_update_view(request, pk):
             user = CustomUser.objects.get(pk=pk)
             if name: user.name = name
             if email: user.email = email
-            if hasattr(VendorUser.objects.get(user=user), 'is_vendor'):
-                vendor = VendorUser.objects.get(user=user)
-                if vendor_name: vendor.vendor_name = vendor_name
-                vendor.save()
+            try:
+                if hasattr(VendorUser.objects.get(user=user), 'is_vendor'):
+                    vendor = VendorUser.objects.get(user=user)
+                    if vendor_name: vendor.vendor_name = vendor_name
+                    vendor.save()
+                # end if
+            except VendorUser.DoesNotExist:
+                user.save()
             user.save() 
```

Comment:
why are there 2 user.save() here?

Reply:
coz when user is not a vendor, exception will be thrown, then i will need to save the normal user details. the other user.save is for vendors.